src/lib.rs

```diff
@@ -100,7 +100,7 @@ impl BuildHasher for DefaultHashBuilder {
 // Node is an internal struct used to encapsulate the nodes that will be added and
 // removed from `HashRing`
 #[derive(Debug)]
-struct Node<T> {
+pub struct Node<T> {
```

Comment:
Do we need to make this a public type? If we map `Node<T>` to `T` in the `next` method would that allow us to keep the type private? For example:

```rust
pub struct HashRingIterator<T> {
    ring: std::vec::IntoIter<Node<T>>,
}

impl<T> Iterator for HashRingIterator<T> {
    type Item = T;

    fn next(&mut self) -> Option<Self::Item> {
        self.ring.next().map(|node| node.node)
    }
}

impl<T: Clone> IntoIterator for HashRing<T> {
    type Item = T;

    type IntoIter = HashRingIterator<T>;

    fn into_iter(self) -> Self::IntoIter {
        HashRingIterator {
            ring: self.ring.into_iter(),
        }
    }
}
```

Reply:
Ah this is much simpler and works. This is what I was trying to figure out but was unable to. Changes coming soon. Thanks for the assist

Reply:
Ah I missed the Node<T> addition in HashRingIterator. This works great, commit coming soon